src/App.tsx

```diff
@@ -1,39 +1,68 @@
-import React from 'react';
+import { Component } from 'react';
+import { Clock } from './Clock';
 import './App.scss';
 
+type State = {
+  hasClock: boolean,
+  clockName: string,
+  previousClockName: string,
+};
+
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+    previousClockName: '',
```

Comment:
There is no need to add previousClockName to the state. Think the other ways of how can you get it